ipn/ipnlocal/profiles.go

```diff
@@ -382,6 +383,24 @@ func (pm *profileManager) DeleteProfile(id ipn.ProfileID) error {
 	return pm.writeKnownProfiles()
 }
 
+// DeleteAllProfiles removes all known profiles and switches to a new empty
+// profile.
+func (pm *profileManager) DeleteAllProfiles() error {
+	metricDeleteAllProfile.Add(1)
+
+	for _, kp := range pm.knownProfiles {
+		if err := pm.store.WriteState(kp.Key, nil); err != nil {
+			// Write profiles we've already deleted, but return the original
+			// error.
+			pm.writeKnownProfiles()
```

Comment:
why? this wouldn't have changed yet. If you want that, you need to call `delete(pm.knownProfiles, k)`
You also probably want to do pm.NewProfile first

Reply:
Was implementing feedback from a previous iteration of this code (https://github.com/tailscale/tailscale/pull/6434#discussion_r1028674287). I can just do the return, I don't think we can really leave things in a consistent state either way.

Reply:
Oh, yeah not leaving them in an inconsistent state sgtm.
But you still need to call `delete(pm.knownProfiles, k)` otherwise you will end up writing the same contents.

Reply:
I added something like this (it's a bit more awkward, because we want to `delete()` all of the profile entries that we are able to successfully remove from the state store, thus we need to keep track of that).